File: bench/programs/sieve/sieve.js
function run() {
  const n = 2000000;
  const sieve = new Uint8Array(n).fill(1);
  let count = 0;
  for (let p = 2; p < n; p++) {
    if (sieve[p]) {
      count++;
      for (let m = p * 2; m < n; m += p) sieve[m] = 0;
    }
  }
  return count;
}

run(); // warmup the JIT
const t0 = process.hrtime.bigint();
const ans = run();
const dt = process.hrtime.bigint() - t0;
console.log(ans.toString());
console.log(dt.toString());

File: bench/programs/sieve/sieve.rs
use std::hint::black_box;
use std::time::Instant;

fn run() -> i64 {
    let n = black_box(2_000_000usize);
    let mut sieve = vec![true; n];
    let mut count = 0i64;
    let mut p = 2usize;
    while p < n {
        if sieve[p] {
            count += 1;
            let mut m = p * 2;
            while m < n {
                sieve[m] = false;
                m += p;
            }
        }
        p += 1;
    }
    count
}

fn main() {
    black_box(run()); // warmup
    let t0 = Instant::now();
    let ans = black_box(run());
    let dt = t0.elapsed().as_nanos() as i64;
    println!("{}", ans);
    println!("{}", dt);
}
